### PR TITLE
fix: record applied login rules in github login event

### PR DESCRIPTION
**Pull request by nklaassen**
I forgot to add this for GitHub logins when I added it for okta and saml logins in https://github.com/gravitational/teleport.e/pull/1428

### CODE REVIEW

**Review by avatus — APPROVED** [review]

File: lib/auth/github.go [hunk 1]
```diff
@@ -371,6 +371,7 @@ func validateGithubAuthCallbackHelper(ctx context.Context, m githubManager, diag
 
 	auth, err := m.validateGithubAuthCallback(ctx, diagCtx, q)
 	diagCtx.Info.Error = trace.UserMessage(err)
+	event.AppliedLoginRules = diagCtx.Info.AppliedLoginRules
```
Comment (codingllama) [inline lib/auth/github.go:374]: nit: Move this to the struct initialization above?

```diff
	event := &apievents.UserLogin{
		Metadata: apievents.Metadata{
			Type: events.UserLoginEvent,
		},
		Method:            events.LoginMethodGithub,
+		AppliedLoginRules: diagCtx.Info.AppliedLoginRules,
	}
```

**Review by codingllama — APPROVED** [review]

File: lib/auth/github.go (hunk 1 above)
Comment (nklaassen) [reply lib/auth/github.go:374]: so this is kind of ugly but `diagCtx.Info.AppliedLoginRules` doesn't get set until the `m.validateGithubAuthCallback` call

Comment (codingllama) [reply lib/auth/github.go:374]: Ohh, could you sneak in a comment into some other PR saying that, if you can? That's quite surprising.

Comment (nklaassen) [reply lib/auth/github.go:374]: I agree it smells, but basically this entire function copies from `diagCtx` to set fields in the event, I don't think this instance really needs a special comment for it

**Pull request merged**